From ede0dd6448fc9e58d913cc4e57d2d5f531af899f Mon Sep 17 00:00:00 2001
From: Nick Hainke <vincent@systemli.org>
Date: Sun, 1 Mar 2026 17:31:14 +0100
Subject: [PATCH] PASN: Fix the compilation errors without CONFIG_PASN

Fix the compilation errors by protecting the access to pasn in
wpas_request_connection for the CONFIG_PASN disabled case.

Fixes: 5f3a47346be5 ("EPPKE: EPPKE authentication for SME-in-wpa_supplicant case")
Signed-off-by: Nick Hainke <vincent@systemli.org>
---
 wpa_supplicant/wpa_supplicant.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -9516,7 +9516,9 @@ void wpas_request_connection(struct wpa_
 	wpa_s->disconnected = 0;
 	wpa_s->reassociate = 1;
 	wpa_s->last_owe_group = 0;
+#ifdef CONFIG_PASN
 	wpa_pasn_reset(&wpa_s->pasn);
+#endif /* CONFIG_PASN */
 
 	if (wpa_supplicant_fast_associate(wpa_s) != 1)
 		wpa_supplicant_req_scan(wpa_s, 0, 0);
